flink-kubernetes-operator-autoscaler/src/test/java/org/apache/flink/kubernetes/operator/autoscaler/BacklogBasedScalingTest.java

```diff
@@ -100,6 +100,7 @@ public void setup() {
         defaultConf.set(AutoScalerOptions.CATCH_UP_DURATION, Duration.ofSeconds(2));
         defaultConf.set(AutoScalerOptions.SCALING_ENABLED, true);
         defaultConf.set(AutoScalerOptions.MAX_SCALE_DOWN_FACTOR, 1.);
+        defaultConf.set(AutoScalerOptions.MAX_SCALE_UP_FACTOR, Double.MAX_VALUE);
```

Comment:
In my tests, MAX_VALUE might be problematic because:

```
> Double.MAX_VALUE * 2
Infinity
```

Reply:
I think we can set the default value to the same as that of vertex.max-parallelism: Integer.MAX_VALUE. This would avoid overflow issues while effectively allowing a job to scale up from 1 to the max configured parallelism.